mmcls/models/utils/augment/cutmix.py

```diff
@@ -0,0 +1,143 @@
+from abc import ABCMeta, abstractmethod
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+
+from .builder import AUGMENT
+
+
+class BaseCutMixLayer(object, metaclass=ABCMeta):
+    """Base class for CutMixLayer.
+
+    Args:
+        alpha (float): Parameters for Beta distribution. Positive(>0).
+        num_classes (int): The number of classes.
+        prob (float): MixUp probability. It should be in range [0, 1].
+            Default to 1.0
+        cutmix_minmax (List[float], optional): cutmix min/max image ratio,
+            cutmix is active and uses this vs alpha if not None.
+        correct_lam (bool): apply lambda correction when cutmix bbox
+            clipped by image borders. Default to True
+    """
+
+    def __init__(self,
+                 alpha,
+                 num_classes,
+                 prob=1.0,
+                 cutmix_minmax=None,
+                 correct_lam=True):
+        super(BaseCutMixLayer, self).__init__()
+
+        assert isinstance(alpha, float) and alpha > 0
+        assert isinstance(num_classes, int)
+        assert isinstance(prob, float) and 0.0 <= prob <= 1.0
+
+        self.alpha = alpha
+        self.num_classes = num_classes
+        self.prob = prob
+        self.cutmix_minmax = cutmix_minmax
+        self.correct_lam = correct_lam
+
+    def rand_bbox_minmax(self, img_shape, minmax, count=None):
+        """Min-Max CutMix bounding-box Inspired by Darknet cutmix impl,
+        generates a random rectangular bbox based on min/max percent values
+        applied to each dimension of the input image.
+
+        Typical defaults for minmax are usually in the  .2-.3 for min and
+        .8-.9 range for max.
+
+        Args:
+            img_shape (tuple): Image shape as tuple
```

Comment:
Can we restrict the tuple length here? i.e., len(img_shape) == 2. And inform that the first number stands for height and the second one is width.

Reply:
imgs in pytorch network are  in BCHW format, so , perhaps, we don't have to restrict the tuple length here.